// @q-dev/gdk-sdk/lib/contracts/DAO/DAOParameterStorageInstance.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DAOParameterStorageInstance = void 0;
const SystemContractInstance_1 = require("../SystemContractInstance");
class DAOParameterStorageInstance extends SystemContractInstance_1.SystemContractInstance {
    constructor(signer, address) {
        super(signer, "DAOParameterStorage.json", address);
    }
    async setDAOParameter(parameter, txOptions) {
        return this.submitTransaction("setDAOParameter", [parameter], txOptions);
    }
    async setDAOParameters(parameters, txOptions) {
        return this.submitTransaction("setDAOParameters", [parameters], txOptions);
    }
    async removeDAOParameter(parameterName, txOptions) {
        return this.submitTransaction("removeDAOParameter", [parameterName], txOptions);
    }
    async removeDAOParameters(parameterNames, txOptions) {
        return this.submitTransaction("removeDAOParameters", [parameterNames], txOptions);
    }
    async getDAOParameter(parameterName) {
        return this.instance.getDAOParameter(parameterName);
    }
    async getDAOParameters() {
        return this.instance.getDAOParameters();
    }
}
exports.DAOParameterStorageInstance = DAOParameterStorageInstance;
