// @q-dev/gdk-sdk/lib/contracts/core/MasterDAOFactoryInstance.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MasterDAOFactoryInstance = void 0;
const SystemContractInstance_1 = require("../SystemContractInstance");
const types_1 = require("../../types");
class MasterDAOFactoryInstance extends SystemContractInstance_1.SystemContractInstance {
    constructor(signer, address) {
        super(signer, "MasterDAOFactory.json", address);
    }
    async deployDAO(params, txOptions) {
        return this.submitTransaction("deployDAO", [params], txOptions);
    }
    async deployDAOPanel(dao, params, txOptions) {
        return this.submitTransaction("deployDAOPanel", [dao, params], txOptions);
    }
    async configureVotingSituations(dao, situations, txOptions) {
        return this.submitTransaction("configureVotingSituations", [dao, situations], txOptions);
    }
    async configureVetoGroups(dao, params, txOptions) {
        return this.submitTransaction("configureVetoGroups", [dao, params], txOptions);
    }
    async revokeDAOCreatorRole(dao, txOptions) {
        return this.submitTransaction("revokeDAOCreatorRole", [dao], txOptions);
    }
    getDAORegistryAddressFromTx(deployDAOReceipt) {
        return this.getTransactionEvents(deployDAOReceipt)[types_1.EventNames.DeployedDAORegistry].args.daoRegistryProxy_;
    }
}
exports.MasterDAOFactoryInstance = MasterDAOFactoryInstance;
