// @q-dev/gdk-sdk/lib/contracts/core/MasterAccessManagementInstance.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MasterAccessManagementInstance = void 0;
const SystemContractInstance_1 = require("../SystemContractInstance");
class MasterAccessManagementInstance extends SystemContractInstance_1.SystemContractInstance {
    constructor(signer, address) {
        super(signer, "MasterAccessManagement.json", address);
    }
    async grantRoles(to, rolesToGrant, txOptions) {
        return this.submitTransaction("grantRoles", [to, rolesToGrant], txOptions);
    }
    async revokeRoles(from, rolesToRevoke, txOptions) {
        return this.submitTransaction("revokeRoles", [from, rolesToRevoke], txOptions);
    }
    async addPermissionsToRole(role, permissionsToAdd, allowed, txOptions) {
        return this.submitTransaction("addPermissionsToRole", [role, permissionsToAdd, allowed], txOptions);
    }
    async removePermissionsFromRole(role, permissionsToRemove, allowed, txOptions) {
        return this.submitTransaction("removePermissionsFromRole", [role, permissionsToRemove, allowed], txOptions);
    }
    async getUserRoles(who) {
        return this.instance.getUserRoles(who);
    }
    async getRolePermissions(role) {
        return this.instance.getRolePermissions(role);
    }
}
exports.MasterAccessManagementInstance = MasterAccessManagementInstance;
