// @openzeppelin/hardhat-upgrades/dist/utils/etherscan-api.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.verifyAndGetStatus = exports.RESPONSE_OK = exports.getEtherscanAPIConfig = exports.callEtherscanApi = void 0;
const EtherscanVerifyContractRequest_1 = require("./hardhat-etherscan/EtherscanVerifyContractRequest");
const EtherscanService_1 = require("./hardhat-etherscan/EtherscanService");
const resolveEtherscanApiKey_1 = require("./hardhat-etherscan/resolveEtherscanApiKey");
const upgrades_core_1 = require("@openzeppelin/upgrades-core");
const undici_1 = require("undici");
const debug_1 = __importDefault(require("./debug"));
const chain_config_1 = require("@nomicfoundation/hardhat-verify/dist/src/chain-config");
/**
 * Call the configured Etherscan API with the given parameters.
 *
 * @param etherscanApi The Etherscan API config
 * @param params The API parameters to call with
 * @returns The Etherscan API response
 */
async function callEtherscanApi(etherscanApi, params) {
    const parameters = new URLSearchParams({ ...params, apikey: etherscanApi.key });
    const response = await (0, undici_1.request)(etherscanApi.url, {
        method: 'POST',
        headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
        body: parameters.toString(),
    });
    if (!(response.statusCode >= 200 && response.statusCode <= 299)) {
        const responseBodyText = await response.body.text();
        throw new upgrades_core_1.UpgradesError(`Etherscan API call failed with status ${response.statusCode}, response: ${responseBodyText}`);
    }
    const responseBodyJson = await response.body.json();
    (0, debug_1.default)('Etherscan response', JSON.stringify(responseBodyJson));
    return responseBodyJson;
}
exports.callEtherscanApi = callEtherscanApi;
/**
 * Gets the Etherscan API parameters from Hardhat config.
 * Throws an error if Etherscan API key is not present in config.
 */
async function getEtherscanAPIConfig(hre) {
    const etherscanConfig = hre.config.etherscan; // This should never be undefined, but check just in case
    const endpoints = await (0, chain_config_1.getCurrentChainConfig)(hre.network, etherscanConfig ? etherscanConfig.customChains : []);
    const key = (0, resolveEtherscanApiKey_1.resolveEtherscanApiKey)(etherscanConfig?.apiKey, endpoints.network);
    return { key, url: endpoints.urls.apiURL };
}
exports.getEtherscanAPIConfig = getEtherscanAPIConfig;
exports.RESPONSE_OK = '1';
async function verifyAndGetStatus(params, etherscanApi) {
    const request = (0, EtherscanVerifyContractRequest_1.toVerifyRequest)(params);
    const response = await (0, EtherscanService_1.verifyContract)(etherscanApi.url, request);
    const statusRequest = (0, EtherscanVerifyContractRequest_1.toCheckStatusRequest)({
        apiKey: etherscanApi.key,
        guid: response.message,
    });
    const status = await (0, EtherscanService_1.getVerificationStatus)(etherscanApi.url, statusRequest);
    return status;
}
exports.verifyAndGetStatus = verifyAndGetStatus;
