// @q-dev/gdk-sdk/lib/utils/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ETHEREUM_ADDRESS = exports.EmptyBytes32 = exports.HALF_PERCENTAGE = exports.ZERO_ADDRESS = exports.DefaultQRC721Params = exports.DefaultQRC20Params = exports.DAO_PANEL_LIMIT_NAME = exports.MASTER_ROLE = exports.MASTER_DAO_REGISTRY_DEP = exports.MASTER_DAO_FACTORY_DEP = exports.VOTING_REGISTRY_DEP = exports.VOTING_FACTORY_DEP = exports.TOKEN_FACTORY_DEP = exports.TOKEN_REGISTRY_DEP = exports.GENERAL_VOTING_NAME = exports.EXPERTS_VOTING_NAME = exports.QRC721_NAME = exports.QRC20_NAME = exports.DAO_REG_PARAMETER_STORAGE_NAME = exports.DAO_CONF_PARAMETER_STORAGE_NAME = exports.DAO_PARAMETER_STORAGE_NAME = exports.DAO_MEMBER_STORAGE_NAME = exports.DAO_VOTING_REGISTRY_NAME = exports.DAO_VOTING_FACTORY_NAME = exports.DAO_TOKEN_REGISTRY_NAME = exports.DAO_TOKEN_FACTORY_NAME = exports.DAO_PERMISSION_MANAGER_NAME = exports.DAO_REGISTRY_NAME = exports.DAO_VAULT_NAME = exports.DAO_RESERVED_NAME = exports.MASTER_DAO_REGISTRY = exports.MASTER_DAO_FACTORY = exports.EXPERT_PERMISSION = exports.VETO_FOR_PERMISSION = exports.VOTE_FOR_PERMISSION = exports.VETO_PERMISSION = exports.VOTE_PERMISSION = exports.CREATE_VOTING_PERMISSION = exports.CHANGE_METADATA_PERMISSION = exports.RECEIVE_PERMISSION = exports.SPEND_PERMISSION = exports.BURN_PERMISSION = exports.MINT_PERMISSION = exports.EXECUTE_PERMISSION = exports.DELETE_PERMISSION = exports.UPDATE_PERMISSION = exports.CREATE_PERMISSION = void 0;
exports.CREATE_PERMISSION = "CREATE";
exports.UPDATE_PERMISSION = "UPDATE";
exports.DELETE_PERMISSION = "DELETE";
exports.EXECUTE_PERMISSION = "EXECUTE";
exports.MINT_PERMISSION = "MINT";
exports.BURN_PERMISSION = "BURN";
exports.SPEND_PERMISSION = "SPEND";
exports.RECEIVE_PERMISSION = "RECEIVE";
exports.CHANGE_METADATA_PERMISSION = "CHANGE_METADATA";
exports.CREATE_VOTING_PERMISSION = "CREATE_VOTING";
exports.VOTE_PERMISSION = "VOTE";
exports.VETO_PERMISSION = "VETO";
exports.VOTE_FOR_PERMISSION = "VOTE_FOR";
exports.VETO_FOR_PERMISSION = "VETO_FOR";
exports.EXPERT_PERMISSION = "EXPERT";
exports.MASTER_DAO_FACTORY = "MASTER_DAO_FACTORY";
exports.MASTER_DAO_REGISTRY = "MASTER_DAO_REGISTRY";
exports.DAO_RESERVED_NAME = "DAO Token Holder";
exports.DAO_VAULT_NAME = "DAO_VAULT";
exports.DAO_REGISTRY_NAME = "DAO_REGISTRY";
exports.DAO_PERMISSION_MANAGER_NAME = "DAO_PERMISSION_MANAGER";
exports.DAO_TOKEN_FACTORY_NAME = "DAO_TOKEN_FACTORY";
exports.DAO_TOKEN_REGISTRY_NAME = "DAO_TOKEN_REGISTRY";
exports.DAO_VOTING_FACTORY_NAME = "DAO_VOTING_FACTORY";
exports.DAO_VOTING_REGISTRY_NAME = "DAO_VOTING_REGISTRY";
exports.DAO_MEMBER_STORAGE_NAME = "DAO_MEMBER_STORAGE";
// Used to get the implementation from the master DAO Registry
exports.DAO_PARAMETER_STORAGE_NAME = "DAO_PARAMETER_STORAGE";
// Parameter storage for configuration parameters such as votingPeriod, vetoPeriod, etc.
exports.DAO_CONF_PARAMETER_STORAGE_NAME = "DAO_CONF_PARAMETER_STORAGE";
// Parameter storage for regular experts parameters
exports.DAO_REG_PARAMETER_STORAGE_NAME = "DAO_REG_PARAMETER_STORAGE";
exports.QRC20_NAME = "QRC20";
exports.QRC721_NAME = "QRC721";
exports.EXPERTS_VOTING_NAME = "EXPERTS_VOTING";
exports.GENERAL_VOTING_NAME = "GENERAL_VOTING";
exports.TOKEN_REGISTRY_DEP = "TOKEN_REGISTRY";
exports.TOKEN_FACTORY_DEP = "TOKEN_FACTORY";
exports.VOTING_FACTORY_DEP = "VOTING_FACTORY";
exports.VOTING_REGISTRY_DEP = "VOTING_REGISTRY";
exports.MASTER_DAO_FACTORY_DEP = "MASTER_DAO_FACTORY";
exports.MASTER_DAO_REGISTRY_DEP = "MASTER_DAO_REGISTRY";
exports.MASTER_ROLE = "MASTER";
exports.DAO_PANEL_LIMIT_NAME = "constitution.maxPanelPerDAO";
exports.DefaultQRC20Params = {
    name: "name",
    symbol: "symbol",
    contractURI: "URI",
    decimals: 18,
    totalSupplyCap: 0,
};
exports.DefaultQRC721Params = {
    name: "name",
    symbol: "symbol",
    contractURI: "URI",
    baseURI: "BASE_URI",
    totalSupplyCap: 0,
};
exports.ZERO_ADDRESS = "0x0000000000000000000000000000000000000000";
exports.HALF_PERCENTAGE = "500000000000000000000000000";
exports.EmptyBytes32 = "0x0000000000000000000000000000000000000000000000000000000000000000";
exports.ETHEREUM_ADDRESS = "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE";
