// @q-dev/gdk-sdk/lib/contracts/RBACInstance.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RBACInstance = void 0;
const SystemContractInstance_1 = require("./SystemContractInstance");
class RBACInstance extends SystemContractInstance_1.SystemContractInstance {
    constructor(signer, abiFile, address) {
        super(signer, abiFile, address);
        this.address = address;
    }
    async grantRoles(to, rolesToGrant, txOptions) {
        return this.submitTransaction("grantRoles", [to, rolesToGrant], txOptions);
    }
    async revokeRoles(from, rolesToRevoke, txOptions) {
        return this.submitTransaction("revokeRoles", [from, rolesToRevoke], txOptions);
    }
    async addPermissionsToRole(role, permissionsToAdd, allowed, txOptions) {
        return this.submitTransaction("addPermissionsToRole", [role, permissionsToAdd, allowed], txOptions);
    }
    async removePermissionsFromRole(role, permissionsToRemove, allowed, txOptions) {
        return this.submitTransaction("removePermissionsFromRole", [role, permissionsToRemove, allowed], txOptions);
    }
    async getUserRoles(who) {
        return this.instance.getUserRoles(who);
    }
    async getRolePermissions(role) {
        return this.instance.getRolePermissions(role);
    }
}
exports.RBACInstance = RBACInstance;
