// @q-dev/gdk-sdk/lib/contracts/RoleManagedRegistryInstance.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RoleManagedRegistryInstance = void 0;
const SystemContractInstance_1 = require("./SystemContractInstance");
class RoleManagedRegistryInstance extends SystemContractInstance_1.SystemContractInstance {
    constructor(signer, abiFile, address) {
        super(signer, abiFile, address);
        this.address = address;
    }
    async upgradeContract(name, newImplementation, txOptions) {
        return this.submitTransaction("upgradeContract", [name, newImplementation], txOptions);
    }
    async upgradeContractAndCall(name, newImplementation, data, txOptions) {
        return this.submitTransaction("upgradeContractAndCall", [name, newImplementation, data], txOptions);
    }
    async addContract(name, contractAddress, txOptions) {
        return this.submitTransaction("addContract", [name, contractAddress], txOptions);
    }
    async addProxyContract(name, contractAddress, txOptions) {
        return this.submitTransaction("addProxyContract", [name, contractAddress], txOptions);
    }
    async justAddProxyContract(name, contractAddress, txOptions) {
        return this.submitTransaction("justAddProxyContract", [name, contractAddress], txOptions);
    }
    async removeContract(name, txOptions) {
        return this.submitTransaction("removeContract", [name], txOptions);
    }
}
exports.RoleManagedRegistryInstance = RoleManagedRegistryInstance;
