// @q-dev/gdk-sdk/lib/ethers-contracts/factories/DAOMetadata__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DAOMetadata__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        anonymous: false,
        inputs: [
            {
                indexed: false,
                internalType: "string",
                name: "daoURI",
                type: "string",
            },
        ],
        name: "DAOURIChanged",
        type: "event",
    },
    {
        anonymous: false,
        inputs: [
            {
                indexed: false,
                internalType: "uint8",
                name: "version",
                type: "uint8",
            },
        ],
        name: "Initialized",
        type: "event",
    },
    {
        inputs: [],
        name: "daoURI",
        outputs: [
            {
                internalType: "string",
                name: "_daoURI",
                type: "string",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "string",
                name: "daoURI_",
                type: "string",
            },
        ],
        name: "setDAOMetadata",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
];
class DAOMetadata__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.DAOMetadata__factory = DAOMetadata__factory;
DAOMetadata__factory.abi = _abi;
