// @q-dev/gdk-sdk/lib/ethers-contracts/factories/IDAOIntegration__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IDAOIntegration__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [],
        name: "getResourceRecords",
        outputs: [
            {
                components: [
                    {
                        internalType: "string",
                        name: "existingRole",
                        type: "string",
                    },
                    {
                        internalType: "string",
                        name: "resource",
                        type: "string",
                    },
                    {
                        internalType: "string[]",
                        name: "permissions",
                        type: "string[]",
                    },
                ],
                internalType: "struct IDAOIntegration.ResourceRecords[]",
                name: "",
                type: "tuple[]",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
class IDAOIntegration__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.IDAOIntegration__factory = IDAOIntegration__factory;
IDAOIntegration__factory.abi = _abi;
