// @q-dev/gdk-sdk/lib/ethers-contracts/factories/IDAOParameterStorage__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IDAOParameterStorage__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [
            {
                internalType: "address",
                name: "member_",
                type: "address",
            },
            {
                internalType: "string",
                name: "permission_",
                type: "string",
            },
        ],
        name: "checkPermission",
        outputs: [
            {
                internalType: "bool",
                name: "",
                type: "bool",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "string",
                name: "parameterName_",
                type: "string",
            },
        ],
        name: "getDAOParameter",
        outputs: [
            {
                components: [
                    {
                        internalType: "string",
                        name: "name",
                        type: "string",
                    },
                    {
                        internalType: "bytes",
                        name: "value",
                        type: "bytes",
                    },
                    {
                        internalType: "enum ParameterType",
                        name: "solidityType",
                        type: "uint8",
                    },
                ],
                internalType: "struct Parameter",
                name: "",
                type: "tuple",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [],
        name: "getDAOParameters",
        outputs: [
            {
                components: [
                    {
                        internalType: "string",
                        name: "name",
                        type: "string",
                    },
                    {
                        internalType: "bytes",
                        name: "value",
                        type: "bytes",
                    },
                    {
                        internalType: "enum ParameterType",
                        name: "solidityType",
                        type: "uint8",
                    },
                ],
                internalType: "struct Parameter[]",
                name: "",
                type: "tuple[]",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [],
        name: "getResource",
        outputs: [
            {
                internalType: "string",
                name: "",
                type: "string",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "string",
                name: "parameterName_",
                type: "string",
            },
        ],
        name: "removeDAOParameter",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "string[]",
                name: "parameterNames_",
                type: "string[]",
            },
        ],
        name: "removeDAOParameters",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                components: [
                    {
                        internalType: "string",
                        name: "name",
                        type: "string",
                    },
                    {
                        internalType: "bytes",
                        name: "value",
                        type: "bytes",
                    },
                    {
                        internalType: "enum ParameterType",
                        name: "solidityType",
                        type: "uint8",
                    },
                ],
                internalType: "struct Parameter",
                name: "parameter_",
                type: "tuple",
            },
        ],
        name: "setDAOParameter",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                components: [
                    {
                        internalType: "string",
                        name: "name",
                        type: "string",
                    },
                    {
                        internalType: "bytes",
                        name: "value",
                        type: "bytes",
                    },
                    {
                        internalType: "enum ParameterType",
                        name: "solidityType",
                        type: "uint8",
                    },
                ],
                internalType: "struct Parameter[]",
                name: "parameters_",
                type: "tuple[]",
            },
        ],
        name: "setDAOParameters",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
];
class IDAOParameterStorage__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.IDAOParameterStorage__factory = IDAOParameterStorage__factory;
IDAOParameterStorage__factory.abi = _abi;
