// @q-dev/gdk-sdk/lib/ethers-contracts/factories/IERC5484__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IERC5484__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                internalType: "address",
                name: "from",
                type: "address",
            },
            {
                indexed: true,
                internalType: "address",
                name: "to",
                type: "address",
            },
            {
                indexed: true,
                internalType: "uint256",
                name: "tokenId",
                type: "uint256",
            },
            {
                indexed: false,
                internalType: "enum IERC5484.BurnAuth",
                name: "burnAuth",
                type: "uint8",
            },
        ],
        name: "Issued",
        type: "event",
    },
    {
        inputs: [
            {
                internalType: "uint256",
                name: "tokenId_",
                type: "uint256",
            },
        ],
        name: "burnAuth",
        outputs: [
            {
                internalType: "enum IERC5484.BurnAuth",
                name: "",
                type: "uint8",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
class IERC5484__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.IERC5484__factory = IERC5484__factory;
IERC5484__factory.abi = _abi;
