// @q-dev/gdk-sdk/lib/ethers-contracts/factories/IFactoryDAORegistry__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IFactoryDAORegistry__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [
            {
                internalType: "string",
                name: "name_",
                type: "string",
            },
            {
                internalType: "address",
                name: "poolAddress_",
                type: "address",
            },
        ],
        name: "addProxyPool",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "string",
                name: "name_",
                type: "string",
            },
            {
                internalType: "uint256",
                name: "offset_",
                type: "uint256",
            },
            {
                internalType: "uint256",
                name: "limit_",
                type: "uint256",
            },
        ],
        name: "injectDependenciesToExistingPools",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "string",
                name: "name_",
                type: "string",
            },
            {
                internalType: "bytes",
                name: "data_",
                type: "bytes",
            },
            {
                internalType: "uint256",
                name: "offset_",
                type: "uint256",
            },
            {
                internalType: "uint256",
                name: "limit_",
                type: "uint256",
            },
        ],
        name: "injectDependenciesToExistingPoolsWithData",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "string[]",
                name: "names_",
                type: "string[]",
            },
            {
                internalType: "address[]",
                name: "newImplementations_",
                type: "address[]",
            },
        ],
        name: "setNewImplementations",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
];
class IFactoryDAORegistry__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.IFactoryDAORegistry__factory = IFactoryDAORegistry__factory;
IFactoryDAORegistry__factory.abi = _abi;
