// @q-dev/gdk-sdk/lib/ethers-contracts/factories/ITokenFactory__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ITokenFactory__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        anonymous: false,
        inputs: [
            {
                indexed: false,
                internalType: "address",
                name: "token",
                type: "address",
            },
            {
                components: [
                    {
                        internalType: "string",
                        name: "name",
                        type: "string",
                    },
                    {
                        internalType: "string",
                        name: "symbol",
                        type: "string",
                    },
                    {
                        internalType: "string",
                        name: "contractURI",
                        type: "string",
                    },
                    {
                        internalType: "uint8",
                        name: "decimals",
                        type: "uint8",
                    },
                    {
                        internalType: "uint256",
                        name: "totalSupplyCap",
                        type: "uint256",
                    },
                ],
                indexed: false,
                internalType: "struct IQRC20.ConstructorParams",
                name: "",
                type: "tuple",
            },
        ],
        name: "DeployedQRC20",
        type: "event",
    },
    {
        anonymous: false,
        inputs: [
            {
                indexed: false,
                internalType: "address",
                name: "token",
                type: "address",
            },
            {
                components: [
                    {
                        internalType: "string",
                        name: "name",
                        type: "string",
                    },
                    {
                        internalType: "string",
                        name: "symbol",
                        type: "string",
                    },
                    {
                        internalType: "string",
                        name: "contractURI",
                        type: "string",
                    },
                    {
                        internalType: "string",
                        name: "baseURI",
                        type: "string",
                    },
                    {
                        internalType: "uint256",
                        name: "totalSupplyCap",
                        type: "uint256",
                    },
                ],
                indexed: false,
                internalType: "struct IQRC721.ConstructorParams",
                name: "",
                type: "tuple",
            },
        ],
        name: "DeployedQRC721",
        type: "event",
    },
    {
        anonymous: false,
        inputs: [
            {
                indexed: false,
                internalType: "address",
                name: "token",
                type: "address",
            },
            {
                components: [
                    {
                        internalType: "string",
                        name: "name",
                        type: "string",
                    },
                    {
                        internalType: "string",
                        name: "symbol",
                        type: "string",
                    },
                    {
                        internalType: "string",
                        name: "contractURI",
                        type: "string",
                    },
                    {
                        internalType: "string",
                        name: "baseURI",
                        type: "string",
                    },
                    {
                        internalType: "uint256",
                        name: "totalSupplyCap",
                        type: "uint256",
                    },
                ],
                indexed: false,
                internalType: "struct IQSBT.ConstructorParams",
                name: "",
                type: "tuple",
            },
        ],
        name: "DeployedQSBT",
        type: "event",
    },
    {
        inputs: [
            {
                components: [
                    {
                        internalType: "string",
                        name: "name",
                        type: "string",
                    },
                    {
                        internalType: "string",
                        name: "symbol",
                        type: "string",
                    },
                    {
                        internalType: "string",
                        name: "contractURI",
                        type: "string",
                    },
                    {
                        internalType: "uint8",
                        name: "decimals",
                        type: "uint8",
                    },
                    {
                        internalType: "uint256",
                        name: "totalSupplyCap",
                        type: "uint256",
                    },
                ],
                internalType: "struct IQRC20.ConstructorParams",
                name: "params_",
                type: "tuple",
            },
            {
                internalType: "address",
                name: "tokenOwner_",
                type: "address",
            },
        ],
        name: "deployQRC20",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                components: [
                    {
                        internalType: "string",
                        name: "name",
                        type: "string",
                    },
                    {
                        internalType: "string",
                        name: "symbol",
                        type: "string",
                    },
                    {
                        internalType: "string",
                        name: "contractURI",
                        type: "string",
                    },
                    {
                        internalType: "string",
                        name: "baseURI",
                        type: "string",
                    },
                    {
                        internalType: "uint256",
                        name: "totalSupplyCap",
                        type: "uint256",
                    },
                ],
                internalType: "struct IQRC721.ConstructorParams",
                name: "params_",
                type: "tuple",
            },
            {
                internalType: "address",
                name: "tokenOwner_",
                type: "address",
            },
        ],
        name: "deployQRC721",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                components: [
                    {
                        internalType: "string",
                        name: "name",
                        type: "string",
                    },
                    {
                        internalType: "string",
                        name: "symbol",
                        type: "string",
                    },
                    {
                        internalType: "string",
                        name: "contractURI",
                        type: "string",
                    },
                    {
                        internalType: "string",
                        name: "baseURI",
                        type: "string",
                    },
                    {
                        internalType: "uint256",
                        name: "totalSupplyCap",
                        type: "uint256",
                    },
                ],
                internalType: "struct IQSBT.ConstructorParams",
                name: "params_",
                type: "tuple",
            },
            {
                internalType: "address",
                name: "tokenOwner_",
                type: "address",
            },
        ],
        name: "deployQSBT",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "nonpayable",
        type: "function",
    },
];
class ITokenFactory__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.ITokenFactory__factory = ITokenFactory__factory;
ITokenFactory__factory.abi = _abi;
