// @q-dev/gdk-sdk/lib/ethers-contracts/factories/IVotingFactory__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IVotingFactory__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        anonymous: false,
        inputs: [
            {
                indexed: false,
                internalType: "address",
                name: "votingAddress",
                type: "address",
            },
            {
                components: [
                    {
                        internalType: "string",
                        name: "panelName",
                        type: "string",
                    },
                    {
                        internalType: "address",
                        name: "votingToken",
                        type: "address",
                    },
                ],
                indexed: false,
                internalType: "struct IDAOVoting.ConstructorParams",
                name: "",
                type: "tuple",
            },
        ],
        name: "DeployedDAOVoting",
        type: "event",
    },
    {
        inputs: [
            {
                components: [
                    {
                        internalType: "string",
                        name: "panelName",
                        type: "string",
                    },
                    {
                        internalType: "address",
                        name: "votingToken",
                        type: "address",
                    },
                ],
                internalType: "struct IDAOVoting.ConstructorParams",
                name: "params_",
                type: "tuple",
            },
            {
                internalType: "string",
                name: "votingName_",
                type: "string",
            },
        ],
        name: "deployDAOVoting",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "nonpayable",
        type: "function",
    },
];
class IVotingFactory__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.IVotingFactory__factory = IVotingFactory__factory;
IVotingFactory__factory.abi = _abi;
