// @q-dev/gdk-sdk/lib/ethers-contracts/factories/ParameterCodecMock__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ParameterCodecMock__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [
            {
                internalType: "string",
                name: "name",
                type: "string",
            },
            {
                internalType: "enum ParameterType",
                name: "expected",
                type: "uint8",
            },
            {
                internalType: "enum ParameterType",
                name: "actual",
                type: "uint8",
            },
        ],
        name: "InvalidParameterType",
        type: "error",
    },
    {
        inputs: [
            {
                components: [
                    {
                        internalType: "string",
                        name: "name",
                        type: "string",
                    },
                    {
                        internalType: "bytes",
                        name: "value",
                        type: "bytes",
                    },
                    {
                        internalType: "enum ParameterType",
                        name: "solidityType",
                        type: "uint8",
                    },
                ],
                internalType: "struct Parameter",
                name: "value_",
                type: "tuple",
            },
        ],
        name: "add",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                components: [
                    {
                        internalType: "string",
                        name: "name",
                        type: "string",
                    },
                    {
                        internalType: "bytes",
                        name: "value",
                        type: "bytes",
                    },
                    {
                        internalType: "enum ParameterType",
                        name: "solidityType",
                        type: "uint8",
                    },
                ],
                internalType: "struct Parameter",
                name: "parameter_",
                type: "tuple",
            },
        ],
        name: "decodeAddress",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "pure",
        type: "function",
    },
    {
        inputs: [
            {
                components: [
                    {
                        internalType: "string",
                        name: "name",
                        type: "string",
                    },
                    {
                        internalType: "bytes",
                        name: "value",
                        type: "bytes",
                    },
                    {
                        internalType: "enum ParameterType",
                        name: "solidityType",
                        type: "uint8",
                    },
                ],
                internalType: "struct Parameter",
                name: "parameter_",
                type: "tuple",
            },
        ],
        name: "decodeBool",
        outputs: [
            {
                internalType: "bool",
                name: "",
                type: "bool",
            },
        ],
        stateMutability: "pure",
        type: "function",
    },
    {
        inputs: [
            {
                components: [
                    {
                        internalType: "string",
                        name: "name",
                        type: "string",
                    },
                    {
                        internalType: "bytes",
                        name: "value",
                        type: "bytes",
                    },
                    {
                        internalType: "enum ParameterType",
                        name: "solidityType",
                        type: "uint8",
                    },
                ],
                internalType: "struct Parameter",
                name: "parameter_",
                type: "tuple",
            },
        ],
        name: "decodeBytes",
        outputs: [
            {
                internalType: "bytes",
                name: "",
                type: "bytes",
            },
        ],
        stateMutability: "pure",
        type: "function",
    },
    {
        inputs: [
            {
                components: [
                    {
                        internalType: "string",
                        name: "name",
                        type: "string",
                    },
                    {
                        internalType: "bytes",
                        name: "value",
                        type: "bytes",
                    },
                    {
                        internalType: "enum ParameterType",
                        name: "solidityType",
                        type: "uint8",
                    },
                ],
                internalType: "struct Parameter",
                name: "parameter_",
                type: "tuple",
            },
        ],
        name: "decodeString",
        outputs: [
            {
                internalType: "string",
                name: "",
                type: "string",
            },
        ],
        stateMutability: "pure",
        type: "function",
    },
    {
        inputs: [
            {
                components: [
                    {
                        internalType: "string",
                        name: "name",
                        type: "string",
                    },
                    {
                        internalType: "bytes",
                        name: "value",
                        type: "bytes",
                    },
                    {
                        internalType: "enum ParameterType",
                        name: "solidityType",
                        type: "uint8",
                    },
                ],
                internalType: "struct Parameter",
                name: "parameter_",
                type: "tuple",
            },
        ],
        name: "decodeUint256",
        outputs: [
            {
                internalType: "uint256",
                name: "",
                type: "uint256",
            },
        ],
        stateMutability: "pure",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "value_",
                type: "address",
            },
            {
                internalType: "string",
                name: "name_",
                type: "string",
            },
        ],
        name: "encodeAddress",
        outputs: [
            {
                components: [
                    {
                        internalType: "string",
                        name: "name",
                        type: "string",
                    },
                    {
                        internalType: "bytes",
                        name: "value",
                        type: "bytes",
                    },
                    {
                        internalType: "enum ParameterType",
                        name: "solidityType",
                        type: "uint8",
                    },
                ],
                internalType: "struct Parameter",
                name: "",
                type: "tuple",
            },
        ],
        stateMutability: "pure",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "bool",
                name: "value_",
                type: "bool",
            },
            {
                internalType: "string",
                name: "name_",
                type: "string",
            },
        ],
        name: "encodeBool",
        outputs: [
            {
                components: [
                    {
                        internalType: "string",
                        name: "name",
                        type: "string",
                    },
                    {
                        internalType: "bytes",
                        name: "value",
                        type: "bytes",
                    },
                    {
                        internalType: "enum ParameterType",
                        name: "solidityType",
                        type: "uint8",
                    },
                ],
                internalType: "struct Parameter",
                name: "",
                type: "tuple",
            },
        ],
        stateMutability: "pure",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "bytes",
                name: "value_",
                type: "bytes",
            },
            {
                internalType: "string",
                name: "name_",
                type: "string",
            },
        ],
        name: "encodeBytes",
        outputs: [
            {
                components: [
                    {
                        internalType: "string",
                        name: "name",
                        type: "string",
                    },
                    {
                        internalType: "bytes",
                        name: "value",
                        type: "bytes",
                    },
                    {
                        internalType: "enum ParameterType",
                        name: "solidityType",
                        type: "uint8",
                    },
                ],
                internalType: "struct Parameter",
                name: "",
                type: "tuple",
            },
        ],
        stateMutability: "pure",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "string",
                name: "value_",
                type: "string",
            },
            {
                internalType: "string",
                name: "name_",
                type: "string",
            },
        ],
        name: "encodeString",
        outputs: [
            {
                components: [
                    {
                        internalType: "string",
                        name: "name",
                        type: "string",
                    },
                    {
                        internalType: "bytes",
                        name: "value",
                        type: "bytes",
                    },
                    {
                        internalType: "enum ParameterType",
                        name: "solidityType",
                        type: "uint8",
                    },
                ],
                internalType: "struct Parameter",
                name: "",
                type: "tuple",
            },
        ],
        stateMutability: "pure",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "uint256",
                name: "value_",
                type: "uint256",
            },
            {
                internalType: "string",
                name: "name_",
                type: "string",
            },
        ],
        name: "encodeUint256",
        outputs: [
            {
                components: [
                    {
                        internalType: "string",
                        name: "name",
                        type: "string",
                    },
                    {
                        internalType: "bytes",
                        name: "value",
                        type: "bytes",
                    },
                    {
                        internalType: "enum ParameterType",
                        name: "solidityType",
                        type: "uint8",
                    },
                ],
                internalType: "struct Parameter",
                name: "",
                type: "tuple",
            },
        ],
        stateMutability: "pure",
        type: "function",
    },
];
class ParameterCodecMock__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.ParameterCodecMock__factory = ParameterCodecMock__factory;
ParameterCodecMock__factory.abi = _abi;
