// @q-dev/gdk-sdk/lib/ethers-contracts/factories/ParameterCodec__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ParameterCodec__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [
            {
                internalType: "string",
                name: "name",
                type: "string",
            },
            {
                internalType: "enum ParameterType",
                name: "expected",
                type: "uint8",
            },
            {
                internalType: "enum ParameterType",
                name: "actual",
                type: "uint8",
            },
        ],
        name: "InvalidParameterType",
        type: "error",
    },
];
class ParameterCodec__factory {
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.ParameterCodec__factory = ParameterCodec__factory;
ParameterCodec__factory.abi = _abi;
