// @q-dev/gdk-sdk/lib/types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VotingType = exports.ParameterType = exports.EventNames = exports.DefaultVotingSituations = void 0;
var DefaultVotingSituations;
(function (DefaultVotingSituations) {
    DefaultVotingSituations["General"] = "General Update Vote";
    DefaultVotingSituations["ConfigurationParameter"] = "Configuration Parameter Vote";
    DefaultVotingSituations["RegularParameter"] = "Regular Parameter Vote";
    DefaultVotingSituations["Membership"] = "Membership Vote";
    DefaultVotingSituations["Constitution"] = "Constitution Vote";
    DefaultVotingSituations["PermissionManager"] = "Permission Manager Vote";
    DefaultVotingSituations["DAORegistry"] = "DAO Registry Vote";
})(DefaultVotingSituations = exports.DefaultVotingSituations || (exports.DefaultVotingSituations = {}));
var EventNames;
(function (EventNames) {
    EventNames["DeployedDAORegistry"] = "DeployedDAORegistry";
    EventNames["DeployedDAOPanelVoting"] = "DeployedDAOPanelVoting";
    EventNames["DeployedDAOParameterStorage"] = "DeployedDAOParameterStorage";
    EventNames["DeployedDAOMemberStorage"] = "DeployedDAOMemberStorage";
})(EventNames = exports.EventNames || (exports.EventNames = {}));
var ParameterType;
(function (ParameterType) {
    ParameterType[ParameterType["NONE"] = 0] = "NONE";
    ParameterType[ParameterType["ADDRESS"] = 1] = "ADDRESS";
    ParameterType[ParameterType["UINT256"] = 2] = "UINT256";
    ParameterType[ParameterType["STRING"] = 3] = "STRING";
    ParameterType[ParameterType["BYTES"] = 4] = "BYTES";
    ParameterType[ParameterType["BOOL"] = 5] = "BOOL";
})(ParameterType = exports.ParameterType || (exports.ParameterType = {}));
var VotingType;
(function (VotingType) {
    VotingType[VotingType["NonRestricted"] = 0] = "NonRestricted";
    VotingType[VotingType["Restricted"] = 1] = "Restricted";
    VotingType[VotingType["PartiallyRestricted"] = 2] = "PartiallyRestricted";
})(VotingType = exports.VotingType || (exports.VotingType = {}));
